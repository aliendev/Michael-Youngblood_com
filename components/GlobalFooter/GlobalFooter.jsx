import Link from "next/link";
import Container from "..//DesignSystem/Container";
import { Github, TwitterX, Twitch } from "react-bootstrap-icons";

const currentYear = new Date().getFullYear();

export default function GlobalFooter() {
  return (
    <footer className="bg-accent-1 border-t border-accent-2">
      <Container>
        <footer className="d-flex flex-wrap justify-content-between align-items-center py-3 my-4 border-top">
          <div className="d-flex align-items-center">
            <span className="mb-4 mb-md-0 text-muted">
              Copyright &copy; 1982 - {currentYear}{" "}
              <a href="/">Michael Youngblood</a>. All rights reserved.
            </span>
          </div>
          <ul className="nav col-md-3 justify-content-end list-unstyled d-flex">
            <li className="ms-3">
              <Link className="text-muted" href="https://aliendev.com">
                AlienDev.com
              </Link>
            </li>
            <li className="ms-3">
              <Link
                className="text-muted"
                href="https://github.com/aliendev/Michael-Youngblood_com"
              >
                <Github
                  title="Code Repo for Michael-Youngblood.com"
                  className="align-text-top"
                />{" "}
                Code Repo
              </Link>
            </li>
          </ul>
          <ul className="nav col-md-2 justify-content-end list-unstyled d-flex">
            <li className="ms-3">
              <Link className="text-muted" href="https://x.com/aliendev2">
                <TwitterX title="AlienDev on Twitter" />
              </Link>
            </li>
            <li className="ms-3">
              <Link
                className="text-muted"
                href="https://www.twitch.tv/aliendev2"
              >
                <Twitch title="AlienDev on Twitch" />
              </Link>
            </li>
            <li className="ms-3">
              <Link className="text-muted" href="https://github.com/aliendev">
                <Github title="Michael Youngblood on Github" />
              </Link>
            </li>
          </ul>
        </footer>
      </Container>
    </footer>
  );
}
