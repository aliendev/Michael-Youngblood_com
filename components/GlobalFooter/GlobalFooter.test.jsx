import '@testing-library/jest-dom';
import { render, screen } from "@testing-library/react";
import GlobalFooter from "./GlobalFooter";

// check that the footer has the copyright date range
describe('renders copyright date range', () => {
  render(<GlobalFooter />);
  const copyrightText = screen.getByText((content) => content.includes(`Copyright Â© 1982`) && content.includes(new Date().getFullYear().toString()));
  expect(copyrightText).toBeInTheDocument();
});

// check that the footer matches the snapshot
test('renders GlobalFooter unchanged', () => {
  const { container } = render(<GlobalFooter />);
  expect(container).toMatchSnapshot();
})




 



