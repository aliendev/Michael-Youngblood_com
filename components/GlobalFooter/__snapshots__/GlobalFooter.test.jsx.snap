// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GlobalFooter unchanged 1`] = `
<div>
  <footer
    class="bg-accent-1 border-t border-accent-2"
  >
    <div
      class="container mx-auto px-5"
    >
      <footer
        class="align-items-center py-3 my-4 border-top"
      >
        <div
          class="flex flex-row align-items-center pb-3"
        >
          <div
            class="d-flex flex-wrap"
          >
            <ul
              class="nav col list-unstyled p-0 m-0"
            >
              <li
                class="ms-3 p-0 m-0"
              >
                <a
                  class="text-muted"
                  href="https://aliendev.com"
                >
                  AlienDev.com
                </a>
              </li>
              <li
                class="ms-3"
              >
                <a
                  class="text-muted"
                  href="https://github.com/aliendev/Michael-Youngblood_com"
                >
                  <svg
                    alt="Code Repo for Michael-Youngblood.com"
                    class="bi bi-github align-text-top"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27s1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.01 8.01 0 0 0 16 8c0-4.42-3.58-8-8-8"
                    />
                  </svg>
                   
                  Code Repo
                </a>
              </li>
            </ul>
            <ul
              class="d-flex col justify-content-end list-unstyled nav"
            >
              <li
                class="ms-3"
              >
                <a
                  class="text-muted"
                  href="https://x.com/aliendev2"
                >
                  <svg
                    alt="AlienDev on Twitter"
                    class="bi bi-twitter-x"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.6.75h2.454l-5.36 6.142L16 15.25h-4.937l-3.867-5.07-4.425 5.07H.316l5.733-6.57L0 .75h5.063l3.495 4.633L12.601.75Zm-.86 13.028h1.36L4.323 2.145H2.865z"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="ms-3"
              >
                <a
                  class="text-muted"
                  href="https://www.twitch.tv/aliendev2"
                >
                  <svg
                    alt="AlienDev on Twitch"
                    class="bi bi-twitch"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.857 0 1 2.857v10.286h3.429V16l2.857-2.857H9.57L14.714 8V0zm9.714 7.429-2.285 2.285H9l-2 2v-2H4.429V1.143h9.142z"
                    />
                    <path
                      d="M11.857 3.143h-1.143V6.57h1.143zm-3.143 0H7.571V6.57h1.143z"
                    />
                  </svg>
                </a>
              </li>
              <li
                class="ms-3"
              >
                <a
                  class="text-muted"
                  href="https://github.com/aliendev"
                >
                  <svg
                    alt="Michael Youngblood on Github"
                    class="bi bi-github"
                    fill="currentColor"
                    height="1em"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27s1.36.09 2 .27c1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.01 8.01 0 0 0 16 8c0-4.42-3.58-8-8-8"
                    />
                  </svg>
                </a>
              </li>
            </ul>
          </div>
          <div
            class="my-4"
          >
            <p
              class="text-muted"
            >
              Copyright Â© 1982 - 
              2025
               
              <a
                href="/"
              >
                Michael Youngblood
              </a>
              . All rights reserved. 100% Human-Crafted Artisanal Code.
            </p>
          </div>
        </div>
      </footer>
    </div>
  </footer>
</div>
`;
